packages/apps/graph/src/devnet/deployment/helper.ts

```diff
@@ -0,0 +1,72 @@
+import type {
+  ChainId,
+  IClient,
+  ICommand,
+  ICommandResult,
+  IKeyPair,
+  ITransactionDescriptor,
+  IUnsignedCommand,
+} from '@kadena/client';
+import {
+  addSignatures,
+  createClient,
+  isSignedTransaction,
+} from '@kadena/client';
+import { sign } from '@kadena/cryptography-utils';
+import { dotenv } from '@utils/dotenv';
+import { logger } from '@utils/logger';
+
+export interface IAccount {
+  account: string;
+  chainId?: ChainId;
+  keys: IKeyPair[];
+}
+
+const getClient = (): IClient =>
+  createClient(
+    ({ chainId }) =>
+      `${dotenv.NETWORK_HOST}/chainweb/0.0/${dotenv.NETWORK_ID}/chain/${chainId}/pact`,
+  );
+
+export const submit = (tx: ICommand): Promise<ITransactionDescriptor> =>
+  getClient().submit(tx);
+
+export const listen = (tx: ITransactionDescriptor): Promise<ICommandResult> =>
+  getClient().listen(tx);
+
+export const signTransaction =
+  (keyPairs: IKeyPair[]) =>
+  (tx: IUnsignedCommand): IUnsignedCommand | ICommand => {
+    const signedTx = keyPairs.reduce((acc, keyPair) => {
+      const { sig } = sign(acc.cmd, {
+        publicKey: keyPair.publicKey,
+        secretKey: keyPair.secretKey,
+      });
+      if (!sig) throw Error('Failed to sign transaction');
+      return addSignatures(acc, { sig, pubKey: keyPair.publicKey });
+    }, tx);
+    return signedTx;
+  };
```

Comment:
There is a [`createSignWithKeypair` function in `@kadena/client`](https://github.com/kadena-community/kadena.js/blob/1f6cadbee5cbf01a67ca2950d7e3a82f99f07705/packages/libs/client/src/signing/keypair/createSignWithKeypair.ts#L49-L69) that you can use. This can simplify things when you have multiple keys. It creates a "keyring" that you can then use to sign with.

Reply:
Perfect! Indeed it makes it a lot cleaner 👐🏽  Implemented changes